### DISCUSSION TITLE
debugging help?

### DISCUSSION THREAD

**Discussion by raould** [T-Troll/alienfx-tools#63] [Q&A] [answered]
hi!!

Are there any lists of step-by-steps to debug why lighting on a machine might be having problems? It isn't just alienfx-tools, the problem happens with AWCC as well but I was hoping that alienfx-tools could maybe help figure out the problem with this laptop & i could avoid having to use AWCC at all ever.

Alienware 13 R3. AWCC will briefly set the lights at startup & then all the lights will soon go dark. Same sort of thing happens whenever I try to use alienfx-cli.exe, the lights turn on for maybe 1 or 2 seconds then they all go back off.
* What Alienware software should I install? Uninstall? I am worried about conflicts between Alienware vs. alienfx-tools.
* What Alienware drivers/services should I disable? or enable?
* Any suggestions about logs to look at? Windows Event Viewer? I didn't see anything obvious in there when i looked but I don't really know what to look for there.

```
C:\Program Files\AlienFX Tools> .\alienfx-cli.exe high-level status
alienfx-cli v5.0.5
Low-level device ready
Dell API ready
Dell API version 2.2.0.0 detected
Devices found: 1
Device #0, Name: Alienware 13 R3, Type: Notebook, Lights: 7
  Light #0, Name: Keyboard Left, Position Unknown, Color (0,0,0), Brightness: 0
  Light #1, Name: Keyboard Middle Left, Position Unknown, Color (0,0,0), Brightness: 0
  Light #2, Name: Keyboard Middle Right, Position Unknown, Color (0,0,0), Brightness: 0
  Light #3, Name: Keyboard Right, Position Unknown, Color (0,0,0), Brightness: 0
  Light #4, Name: Alienhead, Position Unknown, Color (0,0,0), Brightness: 0
  Light #5, Name: Logo, Position Unknown, Color (0,0,0), Brightness: 0
  Light #6, Name: Touchpad, Position Unknown, Color (0,0,0), Brightness: 0
Done.
```

**Comment by T-Troll** [comment]
Yes, it's AWCC most probably.

It's a simple way to check it - issue `alienfx-cli.exe high-level setall=255,255,0 update loop` command. "loop" prevent cli to stop, so AWCC didn't take control back until you stop it with Ctrl+C. If lights stay until you stop cli - yes, it's AWCC turn it off.

BTW, you can also use low-level API, but for `set-one` and `set-action` commands only - they don't require names defined. Or define names in GUI/probe.

How to stop AWCC:
- If you have Desktop version (R3 should have this one) - it's easy to check, you will have tray icon - right-click it and select "Exit". That's all.
- If you have Metro (UWP) version - close AWCC app, go to "Services" and stop "AWCC Service" (or select option in GUI - it will do it automatically).

You can uninstall AWCC, but in this case you can use low-level only, so you need to define light IDs and names (one time). But you still can use high-level for games using my emulation library.

**Reply by raould** [reply]
Thanks for the suggestions! I did go to the Tray, right click on the AWCC icon and "Exit" ... after that, none of the lights work at all, even briefly. Even with "loop" nothing lights up. It seems like AWCC is required for the lights to "listen" to commands. Amazing how many different ways the Dell/Alienware systems can fail. Very useful to have alienfx to test, it does look like Alienware itself is just broken. Wondering if there's anywhere in logging I can look to see if there's a useful error reported by the Dell system.

**Comment by raould** [comment, marked as answer]
When I restart the laptop, the lights turn on first, then they turn off again after a second, just like when trying to use AWCC. But this happens even before the bios splash screen, so there aren't any Windows drivers running at that time. Which makes me wonder if it is broken hardware instead of anything with software.

**Reply by raould** [reply]
Oh wow, I was just trying to update the BIOS and discovered that, apparently, when the laptop is plugged into AC power then the lights stay on! This is no just "Dim Lighting Zones" or "...in Battery Mode Only" because i tested that and the lights are on but dim, whereas my problem as been that the lights are completely off. Cannot yet tell if I will be able to get the lights to ever stay on when running only from battery.

**Comment by T-Troll** [comment]
At older systems, you can not use high-level mode in cli if awcc not running. Use low-level instead.
Seems like it's dimming or wrong default light colors (black) for battery.
As a quick fix, try to disable/enable "dim lights on battery" at awcc tray menu, as well as apply light profile again.
As a long-term fix - use `alienfx-gui` ^_^
